src/entity.rs

```diff
@@ -0,0 +1,131 @@
+use crate::graphics::entity::EntityCube;
+use crate::humanoid;
+use crate::vector::Vector3;
+
+
+#[derive(Debug, PartialEq)]
+/// Enum for the different types of entity
+pub enum EntityKind {
+    Player
+}
+
+/// Contain the data of an entity
+pub struct Entity {
+    id: usize,
+    entity_type: EntityKind,
+    position: Vector3,
+    orientation: [f32;2],
+}
+
+impl Entity {
+    pub fn new(id: usize, entity_type: EntityKind, position: Vector3, orientation: [f32;2]) -> Self {
+        Self { 
+            id,
+            entity_type,
+            position,
+            orientation,
+        }
+    }
+    
+    pub fn set_position(&mut self, position: Vector3) {
+        self.position = position;
+    }
+    
+    pub fn set_orientation(&mut self, orientation: [f32;2]) {
+        self.orientation = orientation;
+    }
+    
+    pub fn position(&self) -> &Vector3 {
+        &self.position
+    }
+    
+    pub fn orientation(&self) -> [f32; 2] {
+        self.orientation
+    }
+        
+    pub fn id(&self) -> usize {
+        self.id
+    }
+        
+    pub fn entity_type(&self) -> &EntityKind {
+        &self.entity_type
+    }
+
+    /// Draw the entity and return a Vec of EntityCube
+    pub fn draw_entity(&mut self) -> Vec<EntityCube> {
+
+        let ent = match self.entity_type {
+            EntityKind::Player => {
+                humanoid::draw( self.position, self.orientation)
+            }
+        };
+
+        ent
+    }
+
+}
+
+/// Contain all the entities
+pub struct EntityManager {
+    entity_list: Vec<Entity>,
+    n_entity: usize
+}
+
```

Comment:
Ok actually the struct Entity Manager has a lot of problems but I will fix them myself. Just put this class in its own file and we will merge it like this.